### PR TITLE
Keep scouring openrct2-ui to reduce individual {x, y} usage

### PR DESCRIPTION
**Pull request by tupaschoal**
I advise looking it on a commit-per-commit basis

### CODE REVIEW

File: src/openrct2-ui/windows/RideConstruction.cpp [hunk 1]
```diff
@@ -1710,21 +1710,19 @@ static void RideConstructPlacedForwardGameActionCallback(const GameAction* ga, c
...
         if (!(trackDirection & 4))
         {
-            x -= CoordsDirectionDelta[trackDirection].x;
-            y -= CoordsDirectionDelta[trackDirection].y;
+            trackPos -= CoordsDirectionDelta[trackDirection];
         }
 
         CoordsXYE next_track;
-        if (track_block_get_next_from_zero(x, y, z, ride, trackDirection, &next_track, &z, &trackDirection, false))
+        if (track_block_get_next_from_zero(
```
Comment (Gymnasiast) [inline src/openrct2-ui/windows/RideConstruction.cpp:1720]: Also a candidate for refactoring. Probably best to make a note of it for another PR.

Comment (tupaschoal) [reply src/openrct2-ui/windows/RideConstruction.cpp:1720]: Yup, I noted, didn't want to go too far on this PR

File: src/openrct2/ride/Ride.cpp [hunk 2]
```diff
@@ -1147,55 +1147,51 @@ void ride_clear_blocked_tiles(Ride* ride)
...
-        return 1;
+        return std::nullopt;
 
     // Now find all the elements that belong to this track piece
     int32_t sequence = trackElement->GetSequenceIndex();
     uint8_t mapDirection = trackElement->GetDirection();
 
     CoordsXY offsets = { trackBlock[sequence].x, trackBlock[sequence].y };
-    CoordsXY newCoords = { *x, *y };
+    CoordsXY newCoords = { location };
```
Comment (Gymnasiast) [inline src/openrct2/ride/Ride.cpp:1174]: Can you get rid of the braces?

Comment (tupaschoal) [reply src/openrct2/ride/Ride.cpp]: Done.
